src/client/index.ts

```diff
@@ -59,6 +59,25 @@ export const activate: ActivationFunction = (context: RendererContext<void>) =>
           terminalElement.setAttribute('uuid', payload.output['runme.dev/uuid'])
           terminalElement.setAttribute('terminalFontFamily', payload.output.terminalFontFamily)
           terminalElement.setAttribute('terminalFontSize', payload.output.terminalFontSize.toString())
+
+          if (payload.output.removeState) {
+            context.setState()
+          }
+          terminalElement.addEventListener('onNotebookTerminalClosed', (data: CustomEventInit<{ content: string }>) => {
+            const content = data.detail!.content
+            if (content) {
+              context.setState(content)
+            }
+          })
+
+          const decodedNotebookTerminalContent = decodeURIComponent(
+            escape(window.atob(payload.output.content || ''))
+          )
+
+          terminalElement.setAttribute('initialContent',
```

Comment:
Why does this PR contain a bunch of stuff to persist notebook content? It should really use the Notebook Serializer if possible.

Shouldn't we separate maintaining state into a separate PR?